sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -741,10 +866,13 @@ def get_media_val_str(num_logical_ports, lane_dict, logical_idx):
 
 
 def notify_media_setting(logical_port_name, transceiver_dict,
-                         app_port_tbl, port_mapping):
+                         app_port_tbl, cfg_port_tbl, port_mapping):
+
     if not g_dict:
         return
 
+    port_speed, lane_count = get_speed_and_lane_count(logical_port_name, cfg_port_tbl)
```

Comment:
should we continue if port_speed or lane_count is -1?

Reply:
if port_speed or lane_count have invalid values, then lane_speed_key will be None, and therefore, if media_settings.json supports SI values per speed, then get_media_settings_value() will return an empty result, according to the updated flow of the parser. So in the case of invalid port_speed or lane_count, nothing will be posted to APP_DB during notify_media_settings().